indico/modules/events/timetable/controllers/display.py

```diff
@@ -100,7 +103,69 @@ def _process(self):
                                 back_url=url_for('.timetable', self.event))
 
 
+class RHTimetableExportWeasyPrint(RHTimetableProtectionBase):
+    def _process(self):
+        form = TimetablePDFExportForm(formdata=request.args, csrf_enabled=False)
+        if form.validate_on_submit():
+            from flask import render_template
+            from indico.modules.events.timetable.models.entries import TimetableEntry
+            from indico.core.db import db
+            from pathlib import Path
+
+            days = {}
+            for day in self.event.iter_days():
+                entries = (self.event.timetable_entries
+                           .filter(db.cast(TimetableEntry.start_dt.astimezone(self.event.tzinfo), db.Date) == day,
+                                           TimetableEntry.parent_id.is_(None))
+                           .order_by(TimetableEntry.start_dt))
+                days[day] = entries
+
+            now = now_utc()
+            html = render_template('events/timetable/pdf/timetable.html', event=self.event, days=days, now=now,
+                                   show_title=form.other.data['showSpeakerTitle'],
+                                   show_affiliation=form.other.data['showSpeakerAffiliation'],
+                                   show_cover_page=form.document_settings.data['showCoverPage'],
+                                   show_toc=form.document_settings.data['showTableContents'],
+                                   show_session_toc=form.document_settings.data['showSessionTOC'],
+                                   page_size=form.pagesize.data,
+                                   page_offset=form.firstPageNumber.data-1,
+                                   show_contribs=form.visible_entries.data['showContribsAtConfLevel'],
+                                   show_breaks=form.visible_entries.data['showBreaksAtConfLevel'],
+                                   new_page_per_session=form.session_info.data['newPagePerSession'],
+                                   use_session_colors=form.session_info.data['useSessionColorCodes'],
+                                   )
```

Comment:
That is the worst type of formatting that exists in the Python world - I still don't understand why anybody uses it (IIRC think even some stdlib modules have this :/)...

In any case, let's not use that in our codebase! :)

If you wrap after the opening `(` of of the `render_template` call you get something much nicer, at the cost of just one extra line:

```py
html = render_template(
    'events/timetable/pdf/timetable.html',
    event=self.event,
    days=days, 
    now=now,
    show_title=form.other.data['showSpeakerTitle'],
    # ...
)
```